train_mnist.py

```diff
@@ -126,29 +126,15 @@ def scscn(x, num, num_conv):
             l = int(h % (m*n))
             j = int(l / n)
             k = int(l % n)
-            if (j == 0)and(k == 0):
-                rr = False
-            # calculate the output of the convolution fliter
-            output = output.write((i * m + j ) * n + k,
-                        tf.identity(small_cnn(
-                                tf.slice(x, [0, j * stride, k * stride, 0],
-                                    [-1, a, b, -1]),
-                                        num_conv, i, j, k, rr, keep_prob)))
-            rr = True
-    # return the concated and reshaped data of output
-    for i in range(m):
-        for j in range(n):
-            for k in range(num):
-                if (j == 0)and(k == 0)and(i == 0):
-                    output_ = output.read((k * m + i ) * n + j)
-                else:
-                    output_ = tf.concat([output_,
-                                    output.read((k * m + i ) * n + j )], 1)
-    return tf.reshape(avg_pool(tf.reshape(output_,
-                      [-1,
-                      m,
-                      n,
-                      num * num_conv]), 5, 5), [-1, num_conv]), keep_prob
+            slicing = slicing.write(h,
+                        tf.slice(x, [0, j * stride, k * stride, 0],
+                                    [-1, a, b, -1]))
+    with tf.name_scope('scn'):
+        scn_input = slicing.concat()
+        output = small_cnn(scn_input, num_conv, keep_prob)
```

Comment:
This call looks wrong. You may want to fix it.

Reply:
It works on my computer.
These change will rise the calculate speed by 70%.

Reply:
According the definition here:
 `def small_cnn(x, num_conv, id, j, k, reuse, keep_prob):`, 
I don't see how the call `small_cnn(scn_input, num_conv, keep_prob)` may work by any mean. (And it indeed does not on my computer. )

Reply:
I have been fixed in  #5